pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1487,41 +1487,48 @@ public PinotResourceManagerResponse addSegment(final SegmentMetadata segmentMeta
     return res;
   }
 
-  public int reloadAllSegments(String tableName) {
-    int numMessagesSent = 0;
+  public int reloadAllSegments(@Nonnull String tableNameWithType) {
+    LOGGER.info("Sending reload message for table: {}", tableNameWithType);
 
-    List<String> segments = getSegmentsFor(tableName);
-    for (String segmentName : segments) {
-      numMessagesSent += sendSegmentReloadMessage(tableName, segmentName);
-    }
+    Criteria recipientCriteria = new Criteria();
+    recipientCriteria.setRecipientInstanceType(InstanceType.PARTICIPANT);
+    recipientCriteria.setInstanceName("%");
+    recipientCriteria.setResource(tableNameWithType);
+    recipientCriteria.setSessionSpecific(true);
+    SegmentReloadMessage segmentReloadMessage = new SegmentReloadMessage(tableNameWithType, null);
```

Comment:
this one is setting the segment name to null, but the server is checking for segment name to be ""?